app/native/components/ThemedText.tsx

```diff
@@ -0,0 +1,86 @@
+import { Text, type TextProps, StyleSheet } from "react-native";
+
+import { useThemeColor } from "@/hooks/useThemeColor";
+
+export type ThemedTextProps = TextProps & {
+  lightColor?: string;
+  darkColor?: string;
+  type?:
+    | "default"
+    | "title"
+    | "defaultSemiBold"
+    | "subtitle"
+    | "link"
+    | "error"
+    | "button";
+};
+
+export function ThemedText({
+  style,
+  lightColor,
+  darkColor,
+  type = "default",
+  ...rest
+}: ThemedTextProps) {
+  const color = useThemeColor({ light: lightColor, dark: darkColor }, "text");
+
+  return (
+    <Text
+      style={[
+        { color },
+        type === "default" ? styles.default : undefined,
+        type === "title" ? styles.title : undefined,
+        type === "defaultSemiBold" ? styles.defaultSemiBold : undefined,
+        type === "subtitle" ? styles.subtitle : undefined,
+        type === "link" ? styles.link : undefined,
+        type === "error" ? styles.error : undefined,
+        type === "button" ? styles.button : undefined,
+        style,
+      ]}
+      {...rest}
+    />
+  );
+}
+
+const styles = StyleSheet.create({
+  default: {
+    fontSize: 16,
+    lineHeight: 24,
+    fontFamily: "Ubuntu_400Regular",
+  },
+  defaultSemiBold: {
+    fontSize: 16,
+    lineHeight: 24,
+    fontWeight: "500",
+    fontFamily: "Ubuntu_500Medium",
+  },
+  title: {
+    fontSize: 32,
+    fontWeight: "bold",
+    lineHeight: 32,
+    fontFamily: "Ubuntu_700Bold",
+  },
+  subtitle: {
+    fontSize: 20,
+    fontWeight: "bold",
+    fontFamily: "Ubuntu_700Bold",
+  },
```

Comment:
You probably don't want hardcoded pixel test sizes for mobile. Better to use rem. The default of 1rem is 16px, so then fontSize 16px would be 1rem, fontSize 32 would be 2rem, etc.

https://www.sitepoint.com/understanding-and-using-rem-units-in-css/